src/server.cpp

```diff
@@ -2661,6 +2653,32 @@ void Server::sendRequestedMedia(session_t peer_id,
 	}
 }
 
+void Server::stepPendingDynMediaCallbacks(float dtime)
+{
+	MutexAutoLock lock(m_env_mutex);
+
+	for (auto it = m_pending_dyn_media.begin(); it != m_pending_dyn_media.end();) {
+		it->second.expiry_timer -= dtime;
+		bool del = it->second.waiting_players.empty() || it->second.expiry_timer < 0;
```

Comment:
Irrelevant actually (I think), the compiler can convert 0 to float automatically just not any other values.

Reply:
Yeah, sorry for the nitpicking.

Reply:
Did you verify my claim or just believe me? :thinking:

Reply:
Neither, I didn't read precisely.

+0.0f ieee floats are just 0s AFAIK, yes. But this is irrelevant here as the 0 or 0.0f is known at compile time, so the compiler can easily do the int to float cast at compile time. This does also work for literals that are non-zero.
However, if you take a literal that causes the operator overloading resolution stuff to choose a different operator, things can get different. Ie. 0.0 does the same as 0.0f, but 0.3 does not the same as 0.3f.

The reason I suggested the .0f is just readability. => nitpicking :-)

Reply:
Here's what I was thinking of: https://godbolt.org/z/bfavsherq

Compilers are not allowed to automatically convert all *double* values to float, with 1.234 you can see both gcc and clang generate an extra `cvtss2sd` instruction. This is not the case for 1.0 or 0.0 because presumably the float representation is exact.
Now for integer-to-float conversions (what we have here) that's mostly irrelevant because the small integers can be exactly represented as floats, so no extra instructions ever.

Reply:
Yes, that's also what I meant with the 0.3 vs. 0.3f.
If we have a float and an int, the `operator<(float, float)` will be chosen, not `operator<(int, int)`, because of casting rules. Hence the float will not need to be casted.
If we have a float and a double, however, it's `operator<(double, double)`, so technically it needs to convert the float first. However for some constants, ie. 0.0, probably all that can be represented without narrowing in a float, as you said, the float instruction with the double casted to float yields the same outputs, so it can use the single-precision instruction.